lib/matplotlib/rcsetup.py

```diff
@@ -565,6 +565,41 @@ def validate_sketch(s):
         raise ValueError("Expected a (scale, length, randomness) triplet")
 
 
+def validate_path_effects(s):
+    if not s:
+        return []
+    if isinstance(s, str) and s.strip().startswith("("):
+        s = ast.literal_eval(s)
+
+    _validate_name = ValidateInStrings("path.effects.function",
+                                        ["Normal",
+                                         "PathPatchEffect",
+                                         "SimpleLineShadow",
+                                         "SimplePatchShadow",
+                                         "Stroke",
+                                         "TickedStroke",
+                                         "withSimplePatchShadow",
+                                         "withStroke",
+                                         "withTickedStroke"])
+
+    def _validate_dict(d):
+        if not isinstance(d, dict):
+            raise ValueError("Expected a dictionary of keyword arguments")
+        return d
+
+    try:
+        # cast to list for the 1 tuple case
+        s = [s] if isinstance(s[0], str) else s
+        return [pe if getattr(pe, '__module__', "") == 'matplotlib.patheffects'
+                else (_validate_name(pe[0].strip()),
+                      {} if len(pe) < 2 else _validate_dict(pe[1]))
+                      for pe in validate_anylist(s)
```

Comment:
I think the `for` line get moved over to the left so that it is more obvious that this is a list comprehension. I'm also not quite sure under what scenario `pe` won't be the module import?

Reply:
Ah, scratch my confusion about `pe`. Although a comment would still help to clarify things.

Reply:
I misunderstood what `getattr(pe, '__module__', "") == 'matplotlib.patheffects'` meant. But the `for` line should definitely get moved to the left rather than getting lined up with the `else` side of things. And a comment wouldn't hurt, either.